agenta-cli/agenta/sdk/decorators/tracing.py

```diff
@@ -1,111 +1,196 @@
+            - ignore = False -> {"username": "admin", "password": "********"} -> {"username": "admin", "password": "********"}
+            """
+            io = {
+                key: value
+                for key, value in io.items()
                 if key
                 not in (
-                    blacklist
-                    if isinstance(blacklist, list)
-                    else []
-                    if blacklist is False
+                    ignore
+                    if isinstance(ignore, list)
                     else io.keys()
+                    if ignore is True
+                    else []
                 )
             }
 
-        def patch(result):
-            TRACE_DEFAULT_KEY = "__default__"
-
-            outputs = result
+            return io
 
-            # PATCH : if result is not a dict, make it a dict
-            if not isinstance(result, dict):
-                outputs = {TRACE_DEFAULT_KEY: result}
-            else:
-                # PATCH : if result is a legacy dict, clean it up
-                if (
-                    "message" in result.keys()
-                    and "cost" in result.keys()
-                    and "usage" in result.keys()
-                ):
-                    outputs = {TRACE_DEFAULT_KEY: result["message"]}
+        def patch(result: Any) -> Dict[str, Any]:
+            """
+            Patch the result to ensure that it is a dictionary, with a default key when necessary.
 
-                    ag.tracing.store_cost(result["cost"])
-                    ag.tracing.store_usage(result["usage"])
+            Example:
+            - result = "Hello, World!" -> {"__default__": "Hello, World!"}
+            - result = {"message": "Hello, World!", "cost": 0.0, "usage": {}} -> {"__default__": "Hello, World!"}
+            - result = {"message": "Hello, World!"} -> {"message": "Hello, World!"}
+            """
+            outputs = (
+                {instrument.DEFAULT_KEY: result}
+                if not isinstance(result, dict)
+                else (
+                    {instrument.DEFAULT_KEY: result["message"]}
+                    if all(key in result for key in ["message", "cost", "usage"])
+                    else result
+                )
+            )
 
             return outputs
 
         @wraps(func)
         async def async_wrapper(*args, **kwargs):
             async def wrapped_func(*args, **kwargs):
-                with ag.tracing.Context(
-                    name=func.__name__,
-                    input=redact(get_inputs(*args, **kwargs), self.ignore_inputs),
-                    spankind=self.spankind,
-                    config=self.config,
-                ):
-                    result = await func(*args, **kwargs)
+                if not ag.tracing.get_current_span().is_recording():
+                    self.kind = "workflow"
+
+                kind = parse_span_kind(self.kind)
```

Comment:
how is this handled in the case of auto instrumentation

Reply:
In the case of autoinstrumentation, we use semconv to map to node types.
See semconv file.